src/redux/reducers/modalReducer.tsx

```diff
@@ -0,0 +1,20 @@
+import { Action } from 'redux';
+import actionTypes from '../actions/actionTypes';
+
+const initialState = {
+    show: false,
+};
+
+const modalReducer = (state = initialState, action: Action) => {
+    switch (action.type) {
+        case actionTypes.OPEN:
+            state.show = true;
+            return Object.assign({}, state);
```

Comment:
Shorter / easier syntax that doesn't mutate the state
```suggestion
            return {
                ...state,
                show: true
            }
```